internal/etl/extract/geopackage.go

```diff
@@ -0,0 +1,110 @@
+package extract
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"sync"
+
+	"github.com/PDOK/gomagpie/config"
+	t "github.com/PDOK/gomagpie/internal/etl/transform"
+	"github.com/go-spatial/geom"
+	"github.com/jmoiron/sqlx"
+	"github.com/mattn/go-sqlite3"
+)
+
+const (
+	sqliteDriverName = "sqlite3_with_extensions"
+
+	// fid,minx,miny,maxx,maxy,geom_type
+	nrOfStandardFieldsInQuery = 6
+)
+
+var once sync.Once
+
+// Load sqlite (with extensions) once.
+//
+// Extensions are by default expected in /usr/lib. For spatialite you can
+// alternatively/optionally set SPATIALITE_LIBRARY_PATH.
+func loadDriver() {
+	once.Do(func() {
+		spatialite := path.Join(os.Getenv("SPATIALITE_LIBRARY_PATH"), "mod_spatialite")
+		driver := &sqlite3.SQLiteDriver{Extensions: []string{spatialite}}
+		sql.Register(sqliteDriverName, driver)
+	})
+}
+
+type GeoPackage struct {
+	db *sqlx.DB
+}
+
+func NewGeoPackage(path string) (*GeoPackage, error) {
+	loadDriver()
+
+	conn := fmt.Sprintf("file:%s?mode=ro", path)
+	db, err := sqlx.Open(sqliteDriverName, conn)
+	if err != nil {
+		return nil, err
+	}
+	return &GeoPackage{db}, nil
+}
+
+func (g *GeoPackage) Close() {
+	_ = g.db.Close()
+}
+
+func (g *GeoPackage) Extract(table config.FeatureTable, fields []string, limit int, offset int) ([]t.RawRecord, error) {
+	if len(fields) == 0 {
+		return nil, errors.New("no fields provided to read from GeoPackage")
+	}
+
+	query := fmt.Sprintf(`
+		select %[3]s as fid,
+		    st_minx(castautomagic(%[4]s)) as bbox_minx, 
```

Comment:
je kunt `enablegpkgmode()` overwegen
https://kadaster-it.slack.com/archives/C02T4HWN5CZ/p1702376762334029?thread_ts=1702375792.264109&cid=C02T4HWN5CZ

Reply:
check. Heb het nu even bij die casts gehouden. Lijkt verder geen verschil te maken. `enablegpkgmode` maakt het overigens nog steeds niet mogelijk om een st_transform te doen, zie https://gis.stackexchange.com/q/412884. Dat gaan we misschien ook nog nodig hebben (nu een todo gezet).